cli/azd/internal/repository/app_init.go

```diff
@@ -472,23 +485,74 @@ func (i *Initializer) prjConfigFromDetect(
 					config.Resources["postgres"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbPostgres,
 						Name: spec.DbPostgres.DatabaseName,
+						Props: project.PostgresProps{
+							DatabaseName: spec.DbPostgres.DatabaseName,
+							AuthType: decideAuthType(
+								spec.DbPostgres.AuthUsingManagedIdentity,
+								spec.DbPostgres.AuthUsingUsernamePassword,
+								false),
+						},
 					}
 				case appdetect.DbMySql:
 					config.Resources["mysql"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbMySQL,
 						Props: project.MySQLProps{
 							DatabaseName: spec.DbMySql.DatabaseName,
-							AuthType:     "managedIdentity",
+							AuthType: decideAuthType(
+								spec.DbMySql.AuthUsingManagedIdentity,
+								spec.DbMySql.AuthUsingUsernamePassword,
+								false),
 						},
 					}
 				case appdetect.DbRedis:
 					config.Resources["redis"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbRedis,
 					}
+				case appdetect.DbCosmos:
+					cosmosDBProps := project.CosmosDBProps{
+						DatabaseName: spec.DbCosmos.DatabaseName,
+					}
```

Comment:
Is it possible to reuse the struct of Cosmos properties instead of using two: `project.CosmosDBProps` and `spec.DbCosmos`?

Reply:
Let's do the refactor later, I prefer to do it this way to move faster.